backend/controllers/Event.js

```diff
@@ -57,6 +57,26 @@ const createEvent = async (req, res) => {
  * req: request object containing user id, event id, and new event info. For example, new description.
  * res: response object to return status 200.
  */
+const modifyEvent = async (req, res) => {
+  try {
+    const eventId = req.params.id;
+    const { updatedData } = req.body;
+
+    // check if user is the creator? -> might change schema for event
+    // anyone can change for now
```

Comment:
This should be easy to do em. Try to find an EventParticipation document with the provided eventId and userId (@deidaraiorek does your auth middleware give access to user/userId?). If it doesn't exist, then the person should not be able to modify the event.

Reply:
Yes my decoded token function does asign user info to req.
But i think you misunderstood Jenny's idea. I believe she mean that only user who created the event can end it.
Personally I think allowing anyone participated in the event to end it is good enough.